src/main/java/movie/storage/service/mapper/MovieSessionMapper.java

```diff
@@ -0,0 +1,35 @@
+package movie.storage.service.mapper;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import movie.storage.model.MovieSession;
+import movie.storage.model.dto.MovieSessionRequestDto;
+import movie.storage.model.dto.MovieSessionResponseDto;
+import movie.storage.service.CinemaHallService;
+import movie.storage.service.MovieService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+@Component
+public class MovieSessionMapper {
+    private final MovieService movieService;
+    private final CinemaHallService cinemaHallService;
+
+    @Autowired
+    public MovieSessionMapper(MovieService movieService, CinemaHallService cinemaHallService) {
+        this.movieService = movieService;
+        this.cinemaHallService = cinemaHallService;
+    }
+
+    public MovieSession convertDtoToMovieSession(MovieSessionRequestDto movieSessionDto) {
+        return new MovieSession(movieService.getById(movieSessionDto.getMovieId()),
+                cinemaHallService.getById(movieSessionDto.getCinemaHallId()),
+                LocalDateTime.parse(movieSessionDto.getShowTime(),
+                        DateTimeFormatter.ofPattern("d-MM-yyyy hh:mm:ss a")));
+    }
+
+    public MovieSessionResponseDto convertMovieSessionToDto(MovieSession movieSession) {
+        return new MovieSessionResponseDto(movieSession.getCinemaHall().getId(),
```

Comment:
what about movieSessionId?
the order seems to be wrong as well

please return to using Lombok and use setters in mappers

Reply:
Lombok returned. Changed use of constructors to use setters.